### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by mdehoog**
- local geth database support
 - switch to official fastlz implementation
 - use a sliding window for the source of truth

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -83,11 +98,22 @@ func main() {
...
 
+	var nextBlockHash *common.Hash
 	for {
-		block, err := client.BlockByNumber(context.Background(), blockNum)
+		var block *types.Block
+		if nextBlockHash == nil {
+			block, err = client.BlockByNumber(context.Background(), blockNum)
+		} else {
+			block, err = client.BlockByHash(context.Background(), *nextBlockHash)
+		}
```
Comment (mdehoog) [inline main.go:108]: minor optimization: because we don't have to look up the number => hash in the db

File: main.go [hunk 2]
```diff
@@ -391,3 +389,195 @@ func cheap7Estimator(tx []byte) float64 {
 func cheap8Estimator(tx []byte) float64 {
 	return cheapEstimator(tx, 8, 16)
 }
+
+// zlibBatchEstimator simulates a zlib compressor at max compression that works on (large) tx
+// batches.  Should bootstrap it before use by calling it on several samples of representative
```
Comment (roberto-bayardo) [inline main.go:393]: did you want to update this comment? Looks like you don't want bootstrapping in this version where (I think) you're trying to simulate 128kb size batches?

File: main.go [hunk 3]
```diff
@@ -83,11 +97,22 @@ func main() {
...
+		if parentBlockHash == nil {
+			block, err = client.BlockByNumber(context.Background(), blockNum)
+		} else {
+			block, err = client.BlockByHash(context.Background(), *parentBlockHash)
+		}
 		if err != nil {
 			log.Fatal(err)
 		}
+		if block == nil {
+			break
```
Comment (roberto-bayardo) [inline main.go]: would it be better to panic here?    This shouldn't happen unless you're going back before block 0 I think?

File: main.go (hunk 2 above)
Comment (mdehoog) [reply main.go:393]: the comment is still applicable... we need to bootstrap to at least fill up some of the buffer before we start returning responses

**Review by roberto-bayardo — APPROVED** [review]

**Pull request merged**